### ISSUE TITLE
T5 TensorRT model speed reduces with increasing sequences length/batch size becoming slower than unoptimized huggingface model

### ISSUE THREAD

**Issue by VikasOjha666** [NVIDIA/TensorRT#1807]
## Description
I was trying the demo notebook in TensorRT/HuggingFace for T5 optimization. I tried the inference with the Valhalla/t5-small-qa-qg-hl model. While the model is produced 2x speed for smaller lengths. It is slower than the torch model for larger sequences. The accuracy is good. The same trend is being observed with the pretrained t5-small model too. Shouldn't it be faster with larger sequences too?.

## Environment

**TensorRT Version**: 8.4.0.6
**NVIDIA GPU**: Nvidia Tesla K80
**NVIDIA Driver Version**: 
**CUDA Version**: 11.1.105
**CUDNN Version**: 
**Operating System**: Ubuntu 18.04
**Python Version (if applicable)**: 3.7
**Tensorflow Version (if applicable)**: 2.6
**PyTorch Version (if applicable)**: 1.10
**Baremetal or Container (if so, version)**: 

## Steps To Reproduce
Run the demo notebook with the large sequence length

cc @vinhngx @rajeevsrao @ttyio @pbridger @kevinch-nv  @pranavm-nvidia

**Comment by nvpohanh** [comment]
This is a known issue and we are trying to improve the performance for longer sequence lengths. Thanks

**Comment by VikasOjha666** [comment]
@nvpohanh Could you please provide any expected timelines for this?

**Comment by nvpohanh** [comment]
I currently don't have an ETA yet, but I would expect that the upcoming TRT 8.4 release should include some performance improvements compared to TRT 8.4.0.6.

Also, I realized that you were using K80, which has been an outdated GPU. Is it possible that you can try Volta+ GPUs since most of our analysis is done on Volta+ GPUs? Thanks

**Comment by nvpohanh** [comment]
@VikasOjha666 Could you also let us know if you add "--enable-kv-cache" flag? Thanks

**Comment by VikasOjha666** [comment]
@nvpohanh I can try with T4 or P100. I used the tensorRT repo's code to convert the model.Hene there was no way to specify the flag.

**Comment by nvpohanh** [comment]
> I used the tensorRT repo's code to convert the model.

Do you mean using trtexec to generate the engine? If so, could you share the command you use? If not, how did you generate the engine, if you are not using the OSS HuggingFace demo code?

**Comment by VikasOjha666** [comment]
@nvpohanh I used the same huggingface demo code from tensorRT repo to do the conversion.

**Comment by nvpohanh** [comment]
Oh I see. Just to confirm, did you use the command in this page? https://github.com/NVIDIA/TensorRT/blob/main/demo/HuggingFace/notebooks/t5.ipynb

I will also run some local experiments to find out the performance difference between TRT and frameworks (PyTorch) for longer sequence lengths like 512.

**Comment by VikasOjha666** [comment]
@nvpohanh I tried with the older version of this notebook before updation.

**Comment by VikasOjha666** [comment]
@nvpohanh Just tried with new updated notebook with larger sequence length this time the model is faster for longer sequences too. I tested it with batch_size 1 and sequence length up to 432 for t5-base. Seems like the absence of past_key_values i.e. specified by KV was the reason for performance loss for larger sequences as T5 use it in case of larger sequences. Will check later for large batch later as well. Till then thanks a lot for pointing out the updated notebook. Also I noticed in trt.py that the author is also planning to implement torch tensor input bindings with help of data_ptr(). So any idea when it will be done.

**Comment by nvpohanh** [comment]
The HF demo does not support past_key_values (i.e. `--enable-kv-cache` flag) in the TRT path yet. This is currently a work in progress and will be added in future releases.

**Comment by nvpohanh** [comment]
KV-cached support has been added to BART: https://github.com/NVIDIA/TensorRT/tree/main/demo/HuggingFace#how-to-run-with-k-v-cache

It will be migrated to other models as well soon.

**Comment by ttyio** [comment]
closing since no activity for more than 3 weeks, please reopen if you still have question, thanks!

**Comment by VikasOjha666** [comment]
@nvpohanh Any idea about the timelines?

**Comment by nvpohanh** [comment]
kv-cache support has been added to all models in the TRT 8.6.1 GA release.
